pulsar-client-cpp/include/pulsar/Client.h

```diff
@@ -108,6 +111,19 @@ class Client {
     void subscribeAsync(const std::vector<std::string>& topics, const std::string& subscriptionName,
                         const ConsumerConfiguration& conf, SubscribeCallback callback);
 
+    /**
+     * subscribe for multiple topics, which match given regexPattern, under the same namespace.
+     */
+    Result subscribe(const std::string& regexPattern, const std::string& consumerName, Consumer& consumer,
```

Comment:
Instead of using boolean parameter, we could change the method signature into `subscribeWithRegex()` or similar, to make it more explicit in the application code.